File: fix_json.dart
import 'dart:io';

void main() {
  final dir = Directory('lib/i18n');
  final files = dir.listSync().whereType<File>().where((f) => f.path.endsWith('.json'));
  for (final file in files) {
    var data = file.readAsStringSync();
    data = data.replaceAll(r'\${count}', r'${count}');
    data = data.replaceAll(r'\${tier}', r'${tier}');
    data = data.replaceAll(r'\${duration}', r'${duration}');
    data = data.replaceAll(r'\${actor}', r'${actor}');
    file.writeAsStringSync(data);
  }
}

File: fix_json.js
const fs = require('fs');
const path = require('path');

const dir = 'lib/i18n';
const files = fs.readdirSync(dir).filter(f => f.endsWith('.json'));

for (const file of files) {
  const filePath = path.join(dir, file);
  let content = fs.readFileSync(filePath, 'utf8');
  
  // Convert {var} to ${var} for slang compatibility
  content = content.replace(/\{count\}/g, '${count}');
  content = content.replace(/\{tier\}/g, '${tier}');
  content = content.replace(/\{duration\}/g, '${duration}');
  content = content.replace(/\{actor\}/g, '${actor}');
  
  fs.writeFileSync(filePath, content, 'utf8');
}
console.log('Fixed JSON strings successfully.');
